src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java

```diff
@@ -0,0 +1,285 @@
+  private final By closeButtonSelector;
+  private final By progressBarSelector;
+  private final By replayOverlaySelector;
+
+  private boolean mobile;
+
+  public AutoplayVuap(WikiaWebDriver driver, String slot, String adIframeId) {
+    this(driver, slot, By.id(adIframeId), false);
+  }
+
+  public AutoplayVuap(WikiaWebDriver driver, String slot, By adIframeSelector, Boolean mobile) {
+    this.driver = driver;
+    this.wait = new Wait(driver);
+
+    this.slot = slot;
+    this.imaBridgeSelector = By.cssSelector("#" + slot + " .video-player iframe[src*='imasdk']");
+    this.adIframeSelector = adIframeSelector;
+    this.mobile = mobile;
+
+    speakerSelector = By.cssSelector(String.format(SPEAKER_SELECTOR_FORMAT, slot));
+    pauseOverlaySelector = By.cssSelector(String.format(PAUSE_BUTTON_SELECTOR_FORMAT, slot));
+    closeButtonSelector = By.cssSelector(String.format(CLOSE_BUTTON_SELECTOR_FORMAT, slot));
+    progressBarSelector = By.cssSelector(String.format(CURRENT_TIME_SELECTOR_FORMAT, slot));
+    replayOverlaySelector = By.cssSelector(String.format(REPLAY_BUTTON_SELECTOR_FORMAT, slot));
+  }
+
+  public void mute() {
+    if (!isMuted()) {
+      wait.forElementClickable(speakerSelector).click();
+    }
+  }
+
+  public void unmute() {
+    if (isMuted()) {
+      wait.forElementClickable(speakerSelector).click();
+    }
+  }
+
+  public void togglePause() {
+    wait.forElementClickable(pauseOverlaySelector);
+    // hack to click pause overlay on mobile
+    clickByJS(pauseOverlaySelector);
+  }
+
+  public void play() {
+    if (isPausedWithOverlay()) {
+      togglePause();
+    } else if( !this.mobile ) {
+      clickOnArea(2);
+    } else {
+      clickOnArea(4);
+    }
+  }
+
+  private boolean isPausedWithOverlay() {
+    return isVideoPaused() && driver.findElement(pauseOverlaySelector).isDisplayed();
+  }
+
+  private Boolean isVideoPaused() {
+    return isDesktop() ? isDesktopVideoPaused() : isMobileVideoPaused();
+  }
+
+  private boolean isDesktop() {
+    return hasDesktopVideoElement();
+  }
+
+  private Boolean isMobileVideoPaused() {
+    return Boolean.valueOf(driver.findElement(getVideoSelector()).getAttribute("paused"));
+  }
+
+  private Boolean isDesktopVideoPaused() {
+    return Boolean.valueOf(usingImaBridge(webDriver -> {
+      final JavascriptActions js = new JavascriptActions();
+
+      return js.execute("document.querySelector('video') && document.querySelector('video').paused").toString();
+    }));
+  }
+
+  public void replay() {
+    wait.forElementVisible(replayOverlaySelector).click();
+  }
+
+  public void close() {
+    wait.forElementClickable(closeButtonSelector).click();
+  }
+
+  public void closeWithJS() {
+    clickByJS(closeButtonSelector);
+  }
+
+  private void clickByJS(By selector) {
+    new JavascriptActions(driver).execute("arguments[0].click();", driver.findElement(selector));
+  }
+
+  private void clickElementInsideAd(By selector) {
+    usingAdFrame(() -> {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove useless curly braces around statement (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1602)